libc/include/llvm-libc-macros/inttypes-macros.h

```diff
@@ -117,6 +115,94 @@
 #define __UINTPTR_FMTX__ __PRIPTR "X"
 #endif
 
+// only recent clang provides these macros, so we need to check if they are
```

Comment:
Note: using time based adjectives such as "recent" or "modern" in comments doesn't tend to age well.  In 10 years, consider how this comment will have aged.  Instead, it's better to put explicit compiler versions in sources since that's easily grep-able.  That has the benefit that when we raise the minimum required supported compiler versions, we can easily grep for the work arounds previously deployed to support older toolchains, then delete them.